crates/core_simd/src/intrinsics.rs

```diff
@@ -130,6 +130,16 @@ extern "platform-intrinsic" {
     pub(crate) fn simd_reduce_xor<T, U>(x: T) -> U;
 
     // truncate integer vector to bitmask
+    // `fn simd_bitmask(vector) -> unsigned integer` takes a vector mask and
+    // returns the most significant bit (MSB) of each lane in the form
+    // of either:
+    // * an unsigned integer
+    // * an array of `u8`
+    // If the vector has less than 8 lanes, a u8 is returned with zeroed trailing bits.
+    //
+    // The bit order of the result depends on the byte endianness, LSB-first for little
+    // endian and MSB-first for big endian.
+    // Not intended to be called on a vector with values other than 0 and -1.
```

Comment:
Is "Not intended to be called" any different from "it is UB to be called that way"?

If 0 and -1 are the only possible values, the "returns the most significant bit (MSB)" part seems unnecessarily specific -- basically this returns whether each lane is logically `true` or `false`, right?

Reply:
The commentary before that is a barely-trimmed copy of the comments from the implementation.
So at the current moment, it's implemented in such a way that it is **not** UB to do that.
"Not intended to be called" is me adding an ominous suggestion that I am reconsidering whether it should be well-defined or not.

Reply:
I think there's an unwritten contract that the mask arguments must always be valid masks.  We should probably just say "don't do it".  It could certainly end up being UB in another future backend.  All of the intrinsics are poorly documented but I don't think we want to give the impression you can pass weird inputs to them without it being UB.

Reply:
Well, specifically when implementing them in Miri one needs to know what is and is not UB. :)

Reply:
I think so, yeah.

As far as I know Rust is currently against partially init data, so a u8 is either u8 or is MaybeUninit<u8>. I could see us getting uint<4> and then not worrying about the next 4 bits and making `simd_select_bitmask` legal on that, but as far as I understand the current Rust model for uninit data, the entire vector risks becoming uninit if we ever passed a single actually uninit bit into the mask for `simd_select_bitmask`.

Reply:
I checked and we're currently the only actual users of this intrinsic so yeah, we can just define it as UB. Bam.